pkg/drivers/mock/device.go

```diff
@@ -0,0 +1,194 @@
+package mock
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/jumpstarter-dev/jumpstarter/pkg/harness"
+	"github.com/jumpstarter-dev/jumpstarter/pkg/locking"
+)
+
+const JUMPSTARTER_MOCK_PATH = "/tmp/jumpstarter-mock.json"
+
+type MockDeviceData struct {
+	file         *os.File
+	Name         string            `json:"name"`
+	Device       string            `json:"device"`
+	Version      string            `json:"version"`
+	Serial       string            `json:"serial"`
+	Tags         []string          `json:"tags"`
+	Config       map[string]string `json:"config"`
+	Control      map[string]string `json:"control"`
+	Power        string            `json:"power"`
+	ConsoleSpeed int               `json:"console_speed"`
+	UsbConsole   string            `json:"usb_console"`
+	ImagePath    string            `json:"image_path"`
+	ImageOffset  uint64            `json:"image_offset"`
+	Storage      bool              `json:"storage"`
+	Busy         bool              `json:"busy"`
+}
+
+func (d *MockDeviceData) load() error {
+	_, err := d.file.Seek(0, 0)
+	if err != nil {
+		return err
+	}
+	return json.NewDecoder(d.file).Decode(d)
+}
+
+func (d *MockDeviceData) save() error {
+	_, err := d.file.Seek(0, 0)
+	if err != nil {
+		return err
+	}
+	return json.NewEncoder(d.file).Encode(d)
+}
+
+type MockDevice struct {
+	driver   *MockDriver
+	fileLock locking.Lock
+	data     MockDeviceData
+}
+
+func newMockDevice() (*MockDevice, error) {
+	file, err := os.OpenFile(JUMPSTARTER_MOCK_PATH, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		return nil, err
+	}
+	device := &MockDevice{
+		driver: &MockDriver{},
+		data: MockDeviceData{
+			file:         file,
+			Name:         "mock",
+			Device:       "/dev/jumpstarter-mock",
+			Version:      "1.0.0",
+			Serial:       "001",
+			Tags:         []string{},
+			Config:       map[string]string{},
+			Control:      map[string]string{},
+			Power:        "off",
+			ConsoleSpeed: 0,
+			UsbConsole:   "",
+			ImagePath:    "",
+			ImageOffset:  0,
+			Storage:      false,
+			Busy:         false,
+		},
+	}
+	device.data.load() // ignoring error
+	err = device.data.save()
+	if err != nil {
+		return nil, err
+	}
+	return device, nil
+}
+
+func (d *MockDevice) Driver() harness.HarnessDriver {
+	return d.driver
+}
+
+func (d *MockDevice) Version() (string, error) {
+	err := d.data.load() // FIXME: check err
+	return d.data.Version, err
```

Comment:
```suggestion
	err := d.data.load()
	return d.data.Version, err
```

No need for FIXME here, you're already bubbling it up :)